tests/visualization_tests/test_contour.py

```diff
@@ -79,252 +108,443 @@ def fail_objective(_: Trial) -> float:
-    if len(params) == 2:
-        assert figure.data[0]["z"][3][1] == 4.0
-        assert figure.data[0]["z"][2][2] == 2.0
+    study_without_trials = prepare_study_with_trials(no_trials=True)
+    info = _get_contour_info(study_without_trials, params=params)
+    assert len(info.sorted_params) == 0
+    assert len(info.sub_plot_infos) == 0
 
 
 @pytest.mark.parametrize(
     "params",
     [
+        [],
+        ["param_a"],
         ["param_a", "param_b"],
         ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
     ],
 )
-def test_plot_contour_customized_target_name(params: List[str]) -> None:
+def test_get_contour_info_ignored_error(params: Optional[List[str]]) -> None:
+    def fail_objective(_: Trial) -> float:
 
+        raise ValueError
+
+    study = create_study()
+    study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
+    info = _get_contour_info(study, params=params)
+    assert len(info.sorted_params) == 0
+    assert len(info.sub_plot_infos) == 0
+
+
+def test_get_contour_info_error() -> None:
     study = prepare_study_with_trials()
-    figure = plot_contour(study, params=params, target_name="Target Name")
-    for data in figure.data:
-        if "colorbar" in data:
-            assert data["colorbar"]["title"]["text"] == "Target Name"
 
+    with pytest.raises(ValueError):
+        _get_contour_info(study, ["optuna", "Optuna"])
 
-def test_plot_contour_for_few_observations() -> None:
 
-    study = prepare_study_with_trials(less_than_two=True)
-    trials = study.trials
-    reverse_scale = _is_reverse_scale(study, target=None)
+@pytest.mark.parametrize("params", [[], ["param_a"]])
+def test_get_contour_info_empty(params: Optional[List[str]]) -> None:
+    study = prepare_study_with_trials()
+    info = _get_contour_info(study, params=params)
+    assert len(info.sorted_params) <= 1
+    assert len(info.sub_plot_infos) <= 1
 
-    # `x_axis` has one observation.
+
+def test_get_contour_info_2_params() -> None:
     params = ["param_a", "param_b"]
-    info = _get_contour_subplot_info(trials, params[0], params[1], None)
-    contour, scatter = _get_contour_subplot(info, reverse_scale)
-    assert contour.x is None and contour.y is None and scatter.x is None and scatter.y is None
+    study = prepare_study_with_trials()
+    info = _get_contour_info(study, params=params)
+    assert info.sorted_params == params
+    xaxis = info.sub_plot_infos[0][0].xaxis
+    yaxis = info.sub_plot_infos[0][0].yaxis
+    assert xaxis.name == "param_a"
+    assert yaxis.name == "param_b"
+    assert xaxis.range == (0.925, 2.575)
+    assert yaxis.range == (-0.1, 2.1)
+    assert not xaxis.is_log
+    assert not yaxis.is_log
+    assert not xaxis.is_cat
+    assert not yaxis.is_cat
+    assert xaxis.indices == [0.925, 1.0, 2.5, 2.575]
+    assert yaxis.indices == [-0.1, 0.0, 1.0, 2.0, 2.1]
+    assert xaxis.values == [1.0, None, 2.5]
+    assert yaxis.values == [2.0, 0.0, 1.0]
```

Comment:
```suggestion
    assert xaxis == _AxisInfo(name="param_a", range=(0.925, 2.575), is_log=False, is_cat=False, indices=[0.925, 1.0, 2.5, 2.575], values=[1.0, None, 2.5])
    assert yaxis == _AxisInfo(name="param_b", range=(-0.1, 2.1), is_log=False, is_cat=False, indices=[-0.1, 0.0, 1.0, 2.0, 2.1], values=[2.0, 0.0, 1.0])
```
An `_AxisInfo` is a named tuple, so why not simply create a new `_AxisInfo` and compare with it? I think it's more readable that way.

Reply:
> An _AxisInfo is a named tuple, so why not simply create a new _AxisInfo and compare with it? I think it's more readable that way.

I think the current one is fine, since using the method of checking attributes one line at a time, when an error occurs, it is immediately clear which attribute caused the error. If we compare the entire NamedTuple, it would be difficult to determine where the error occurred in a complex NamedTuple.

> And should we add tests for z_values?

Sounds good. I will add it.

Reply:
>  If we compare the entire NamedTuple, it would be difficult to determine where the error occurred in a complex NamedTuple.
I don't think it will be difficult, since pytest reveals all members of named tuples when an error is encountered.

Reply:
Thanks, I will directly compare the NamedTuple.